src/components/blog/posts_list.js

```diff
@@ -1,23 +1,71 @@
 import React from "react"
+import PropTypes from "prop-types"
+import { StaticQuery, graphql } from "gatsby"
 
 import Entry from "./posts_list/entry"
 
 import styles from "./posts_list.module.scss"
 
-const PostsList = () => (
+const renderItem = ({ id, ...entry }) => (
+  <li key={id} className={styles.postsListItem}>
+    <Entry {...entry} />
+  </li>
+)
+
+const BlogPostsList = ({ items }) => (
   <div className={styles.root}>
     <div className={styles.content}>
-      <ol className={styles.postsList}>
-        <li className={styles.postsListItem}>
-          <Entry
-            author="Subvisual"
-            title="Coming soon..."
-            intro="Our blog is making its return. In style."
-          />
-        </li>
-      </ol>
+      <ol className={styles.postsList}>{items.map(renderItem)}</ol>
     </div>
   </div>
 )
 
-export default PostsList
+BlogPostsList.propTypes = {
+  items: PropTypes.array.isRequired,
+}
+
+const query = graphql`
+  {
+    allMarkdownRemark(sort: { order: DESC, fields: [frontmatter___date] }) {
+      nodes {
+        fields {
+          slug
+        }
+        frontmatter {
+          author {
+            name
+          }
+          date
+          id
+          title
+          intro
+        }
+      }
+    }
+  }
+`
+
+export default () => (
+  <StaticQuery
```

Comment:
You can use `useStaticQuery` and not use the horrendous pattern that a render prop is

Reply:
I love you right now :heart:

Reply:
My freelancing sites all use a bunch of hooks sprayed on a `data` folder. 

Images, copy in markdown, everything is there. My components just import those and magically there are images everywhere and pretty copy everywhere. If you know what you are doing Gatsby is bar to none <3